### DISCUSSION TITLE
Buttons in Cards don't work as expected

### DISCUSSION THREAD

**Discussion by MattMacs** [zauberzeug/nicegui#2280] [Q&A] [answered]
### Question

Hi,

I'm just starting with nicegui coming from streamlit (already loving it) 
and currently trying to get give our database application a more user-friendly interface.

I'm building a page with cards in rows, and within the cards I have images, labels and buttons.
What I'm trying to achieve right now, is that when the user clicks the 'Details' button  notification with
the corresponding id pops up.

But regardless which button I click, the notification is always the same (actually from the last card) ?

ui.button('Details', on_click=ui.notify(entry[id]))

Where's my misunderstanding?

**Comment by falkoschindler** [comment, marked as answer]
Hi @MattMacs, this question is answered in our FAQ:
https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-the-last-element-of-my-for-loop-used-in-every-step 🙂

**Comment by MattMacs** [comment]
Hi @falkoschindler,

thanks for the swift reply... works like a charm :)
